neo/rawio/axonarawio.py

```diff
@@ -197,51 +327,165 @@ def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop,
 
         # Read one channel at a time
         raw_signals = np.ndarray(shape=(num_samples,
-                                 len(channel_indexes)),
-                                 dtype=self.data_dtype)
+                                        len(channel_indexes)),
+                                 dtype=bin_dict['data_type'])
 
         for i, ch_idx in enumerate(channel_indexes):
-
             chan_offset = self.channel_memory_offset[ch_idx]
             raw_signals[:, i] = self._raw_signals[sig_ids + chan_offset]
 
         return raw_signals
 
+    def _spike_count(self, block_index, seg_index, unit_index):
+        tetrode_id = unit_index
+
+        raw_spikes = self._raw_spikes[tetrode_id]
+        nb_tetrode_spikes = raw_spikes.shape[0]
+        # also take into account last, potentially incomplete set of spikes
+        nb_unit_spikes = int(np.ceil(nb_tetrode_spikes / 4))
```

Comment:
If `raw_spikes` would be 2d then it would be shape[0] and wowuld avoid this magic //4 number which be a problem when tetrod would become stereotrode for instance.

Reply:
I like the idea of making it 2d. In theory stereotrode data should still have 4 channels:

```
Tetrode and stereotrode data is almost identical, in that both store 4-channels worth of 
data per spike. The only difference is that in stereotrode mode, either channels 3&4 or 
1&2 will be all zeroes, corresponding to a spike on stereotrode 1 or 2, respectively (the 
stereotrode format is very wasteful of file space).
 ```

Reply:
changed to 2D implementation